src/main/java/slimeknights/tconstruct/smeltery/block/FaucetBlock.java

```diff
@@ -0,0 +1,122 @@
+package slimeknights.tconstruct.smeltery.block;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockRenderType;
+import net.minecraft.block.BlockState;
+import net.minecraft.block.ContainerBlock;
+import net.minecraft.client.renderer.RenderType;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.item.BlockItemUseContext;
+import net.minecraft.state.DirectionProperty;
+import net.minecraft.state.StateContainer;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.ActionResultType;
+import net.minecraft.util.Direction;
+import net.minecraft.util.Hand;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.BlockRayTraceResult;
+import net.minecraft.util.math.shapes.ISelectionContext;
+import net.minecraft.util.math.shapes.VoxelShape;
+import net.minecraft.util.math.shapes.VoxelShapes;
+import net.minecraft.world.IBlockReader;
+import net.minecraft.world.World;
+import net.minecraft.world.server.ServerWorld;
+import slimeknights.tconstruct.smeltery.tileentity.FaucetTileEntity;
+
+import javax.annotation.Nullable;
+import java.util.Random;
+
+public class FaucetBlock extends ContainerBlock {
+  public static final DirectionProperty FACING = DirectionProperty.create("facing", (direction) -> {
+    return direction != Direction.DOWN;
+  });
+  protected static final VoxelShape UP_SHAPE = Block.makeCuboidShape(4.0D, 10.0D, 4.0D, 12.0D, 16.0D, 12.0D);
```

Comment:
Make an `EnumMap` with these. Can use a `static {}` block to initialize.

Reply:
Fair enough. I basically did an inverted hopper thing.

Reply:
Think I've handled this in a satisfactory way. Its modeled after how WallTorchBlock
is handled.